src/main/java/emissary/util/ShortNameComparator.java

```diff
@@ -19,34 +19,115 @@
     public int compare(IBaseDataObject obj1, IBaseDataObject obj2) {
         String s1 = obj1.shortName();
         String s2 = obj2.shortName();
+        int index1 = s1.indexOf(Family.SEP, 0);
+        int index2 = s2.indexOf(Family.SEP, 0);
 
-        String[] p1 = s1.split(Family.SEP);
-        String[] p2 = s2.split(Family.SEP);
-
-        for (int i = 1; i < p1.length; i++) {
-
-            if (i >= p2.length) {
+        while (index1 >= 0) {
+            if (index1 >= 0 && index2 < 0) {
```

Comment:
The index1 check is redundant and can be removed (this is checked by the while loop)

Comment: If string 1 has children and string 2 doesn't have children